### ISSUE TITLE
Bug: Wrong `argument` type for `YieldExpression`: `undefined`, but should be `null`

### ISSUE THREAD

**Issue by marvinhagemeister** [typescript-eslint/typescript-eslint#10793]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&showAST=es&fileType=.ts&code=GYVwdgxgLglg9mAVAAmHOAKAlMg3gKGWQE8YBTAGwBMBufAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
function* foo() {
  yield;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

A `YieldExpression` is typed as:

```ts
export interface YieldExpression extends BaseNode {
  type: AST_NODE_TYPES.YieldExpression;
  argument: Expression | undefined; // <-- should be null
  delegate: boolean;
}
```

But the empty value for `argument` is `null` in the playground.

### Actual Result

```json
 {
  "type": "YieldExpression",
  "argument": null,
  "delegate": false,
}        
```

### Versions

see playground link
